evm-contracts/src/v0.6/dev/Flags.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity 0.6.6;
+
+
+import "../Owned.sol";
+import "../dev/SimpleAccessControl.sol";
+
+
+/**
+ * @title The Flags contract
+ * @notice Allows flags to signal to any reader on the access control list.
+ * The owner can set flags, or designate other addresses to set flags. The
+ * owner must turn the flags off, other setters cannot.
+ */
+contract Flags is Owned, SimpleAccessControl {
+
+  mapping(address => bool) private flags;
+  mapping(address => bool) private setters;
+
+  event FlagOn(
+    address indexed subject
+  );
+  event FlagOff(
+    address indexed subject
+  );
+  event SetterEnabled(
+    address indexed setter
+  );
+  event SetterDisabled(
+    address indexed setter
+  );
+
+  /**
+   * @notice read the warning flag status of a contract address.
+   * @param subject The contract address being checked for a warning.
+   */
+  function getFlag(address subject)
+    public
+    view
+    checkAccess()
```

Comment:
It is somewhat strange to guard access to a view function, as it does not modify anything and all data is already public on-chain. It's my conclusion that this flags data is valuable to other contracts on-chain, and that's why we restrict access to it (like a paywall)?

It would be great if you could add a comment explaining why this view function is guarded.

Reply:
There isn't really a specific reason, other than that we think it may be possible we, or someone else, will want it in the future. It's optional now and can be turned off by the owner.

Reply:
Ok, great! I would just add a more explicit comment explaining the reasoning behind it.